src/pages/index.js

```diff
@@ -72,119 +101,118 @@ async function stageNetwork({ testchainId, network }) {
-                }}
-                address={connectedAddress}
-              />
-            }
-            navbar={<Navbar address={connectedAddress} />}
-            sidebar={
-              <Sidebar
-                network={{
-                  id: maker.service('web3').networkId()
-                }}
-                currentAccount={
-                  connectedAddress ? maker.currentAccount() : null
-                }
-                address={connectedAddress}
-              />
-            }
-            content={getPage()}
-          />
-        )
-      );
-    } catch (errMsg) {
-      return <div>{errMsg.toString()}</div>;
+      connectedAddress = maker.currentAddress();
+    } catch (_) {
+      // if no account is connected, we render in read-only mode
     }
-  };
-}
 
-export default createSwitch({
-  paths: {
-    '/': url => {
-      if (networkIsUndefined(url)) return createDefaultNetworkRedirect(url);
-
-      return createPage({
-        title: 'Landing',
-        getContent: withAuthenticatedNetwork(() => <Landing />)
-      });
-    },
-
-    '/overview': url => {
-      if (networkIsUndefined(url)) return createDefaultNetworkRedirect(url);
-
-      return createPage({
-        title: 'Overview',
-        getContent: withAuthenticatedNetwork(() => <Overview />)
-      });
-    },
-
-    '/cdp/:type': url => {
-      if (networkIsUndefined(url)) return createDefaultNetworkRedirect(url);
-      const cdpTypeSlug = url.params.type;
-
-      return createPage({
-        title: 'CDP',
-        getContent: withAuthenticatedNetwork(() => (
-          <CDPPage cdpTypeSlug={cdpTypeSlug} />
-        ))
-      });
+    return (
+      <PageLayout
+        mobileNav={
+          <MobileNav
+            network={{
+              id: maker.service('web3').networkId()
+            }}
+            address={connectedAddress}
+          />
+        }
+        navbar={<Navbar address={connectedAddress} />}
+        sidebar={
+          <Sidebar
+            network={{
+              id: maker.service('web3').networkId()
+            }}
+            currentAccount={connectedAddress ? maker.currentAccount() : null}
+            address={connectedAddress}
+          />
+        }
+      >
+        <View />
+      </PageLayout>
+    );
+  }, route);
+
+const hasNetwork = route =>
+  map((request, context) => {
+    if (networkIsUndefined(request)) {
+      return createDefaultNetworkRedirect(request);
+    } else {
+      return route;
     }
-  }
-});
+  });
 
-function networkIsUndefined(url) {
-  return url.query.network === undefined && url.query.testchainId === undefined;
+export default hasNetwork(
+  compose(
+    withContext(initUserSnap),
+    withContext(networkDetails),
+    withContext(makerContext),
+    withContext(watcherContext),
+    withView((request, context) => (
+      <MakerHooksProvider maker={context.maker}>
+        <View />
+      </MakerHooksProvider>
+    )),
+    withView(
```

Comment:
quoting Navi's doc for `withContext`
`This matcher is useful for providing information that is shared by multiple children — similar to React Context.`

I'm not saying it's the same thing, I'm saying it solves the same problem.

What I really think we should be avoiding right now is creating multiple paths for the same problem solving, since it'll make things less obvious for no added real value.

we can always optimize once it's a good time to optimize, but for now we're doing the v0.1 of the app and we're not moving the needle by changing stuff from `react context` to `react context AND navi context`

Reply:
They do both provide `context`, but for different use-cases:
1. `React.Context` provides context for *components*
2. `withContext` provides context for *navi routes*

Basically if you need context that components themselves don't need but is useful to know within routing (authentication for example), then navi context is preferable.